executor/builder.go

```diff
@@ -5196,9 +5199,37 @@ func (b *executorBuilder) buildCompactTable(v *plannercore.CompactTable) Executo
 		return nil
 	}
 
+	var partitionIDs []int64
+	if v.PartitionNames != nil {
+		if v.TableInfo.Partition == nil {
+			b.err = errors.Errorf("table:%s is not a partition table, but user specify partition name list:%+v", v.TableInfo.Name.O, v.PartitionNames)
+			return nil
+		}
+		// use map to avoid FindPartitionDefinitionByName
+		partitionMap := map[string]int64{}
+		for _, partition := range v.TableInfo.Partition.Definitions {
+			partitionMap[partition.Name.L] = partition.ID
+		}
+
+		for _, partitionName := range v.PartitionNames {
+			partitionID, ok := partitionMap[partitionName.L]
+			if !ok {
+				b.err = table.ErrUnknownPartition.GenWithStackByArgs(partitionName.O, v.TableInfo.Name.O)
+				return nil
+			}
+			partitionIDs = append(partitionIDs, partitionID)
+		}
+	}
+	log.Info("build CompactTableTiFlashExec",
+		zap.String("table", v.TableInfo.Name.O),
+		zap.Int64("table-id", v.TableInfo.ID),
+		zap.String("partition-name", fmt.Sprintf("%v", v.PartitionNames)),
+		zap.String("partition-id", fmt.Sprintf("%v", partitionIDs)),
+	)
```

Comment:
Should remove the debug log?

If you really want to add log, you should add log when executing success or failure.